dsgrid/registry/registry_manager.py

```diff
@@ -221,6 +222,97 @@ def load(
         )
         return cls(params)
 
+    def data_sync(self, project_id, dataset_id, no_prompts=True):
+        """Sync data from the remote dsgrid registry.
+
+        Parameters
+        ----------
+        project_id : str
+            Sync by project_id filter
+        dataset_id : str
+            Sync by dataset_id filter
+        no_prompts :  bool
+            If no_prompts is False, the user will be prompted to continue sync pulling the registry if lock files exist. By default, True.
+        """
+        if not project_id and not dataset_id:
+            raise ValueError("Must provide a dataset_id or project_id for dsgrid data-sync.")
+
+        if project_id:
+            if not self.project_manager.has_id(project_id):
+                raise ValueError(f"No registered project ID = '{project_id}'")
+            project_version = self.project_manager.get_current_version(project_id)
+            config_file = self.project_manager.get_config_file(project_id, project_version)
+            config = ProjectConfig.load(
+                config_file, self.dimension_manager, self.dimension_mapping_manager
+            )
+            if dataset_id:
+                if dataset_id not in config.list_registered_dataset_ids():
+                    raise ValueError(
```

Comment:
It's probably better to raise `DSGValueNotRegistered` so that we can handle all dsgrid exceptions in a common way.